### ISSUE TITLE
Default docker-daemon configuration can fail to find correct DNS

### ISSUE THREAD

**Issue by assafmendelson** [ubuntu/microk8s#252]
Using a default ubuntu installation (e.g. in a VM), ```/etc/resolv.conf``` contains nameserver 127.0.0.53 (which is a proxy). When building docker images, this proxy is not always available.

I have encountered two workarounds:
The first is to fix ```/etc/resolv.conf``` by doing: ```sudo ln -s /run/systemd/resolve/resolv.conf /etc/resolv.conf```

The second (probably better for microk8s) is to configure docker-daemon.json.
Currently it contains:
```
{
  "insecure-registries" : ["localhost:32000"]
}
```
one can find the correct nameserver either by looking at ```/run/systemd/resolve/resolv.conf``` or by doing
```nmcli dev show | grep DNS```.

Assuming the DNS is 1.1.1.1 and 2.2.2.2 then we would change docker-daemon.json as follows:
```
{
  "dns": ["1.1.1.1", "2.2.2.2"],
   "dns-search": ["localdomain"],
  "insecure-registries" : ["localhost:32000"]
}
```

I am not 100% sure where the dns-search element comes from or if it is really needed though (I solved it a while ago and don't quite remember).

Note that this needs to be updated whenever the DNS changes (so it should probably be added/calculated with a proper switch on start, e.g. ```microk8s.start --update_dns```

**Comment by ktsakalozos** [comment]
Hi @assafmendelson 

This seems to be related to the changes in  `/etc/resolv.conf` on 18.04 can you share a case where the problem is always exposed?

Thank you

**Comment by assafmendelson** [comment]
Hi @ktsakalozos 
I have installed xubuntu 18.04 on a VM and then installed microk8s as in the instructions. When doing this various docker usages (e.g. building an image) fail to find the network.

I believe the reason for this is that /etc/resolv.conf points to 127.0.0.53 which is a proxy created by the systemd-resolved service which is unavailable inside the docker network. In any case /run/systemd/resolve/resolv.conf points to the correct network information

**Comment by hacst** [comment]
It looks like this might be fixed with docker 18.09 (see https://github.com/moby/moby/pull/37485). Any idea when we might get this docker version in microk8s?

Currently I'm using the workaround described in https://github.com/docker/libnetwork/issues/2187#issuecomment-413351987

**Comment by ktsakalozos** [comment]
I am not sure when 18.09 will hit the repositories. However, you should know we will stop shipping dockerd and be moving to containerd. This will eliminate the constraint that we have where we do not allow dockerd to be running next to MicroK8s and should also give us better performance.
